src/map/skill.c

```diff
@@ -16321,8 +16332,14 @@ static int skill_check_condition_castbegin(struct map_session_data *sd, uint16 s
 			break;
 		default:
 			if (sd->spiritball < require.spiritball) {
-				clif->skill_fail(sd, skill_id, USESKILL_FAIL_SPIRITS, require.spiritball, 0);
-			return 0;
+				if ((sd->job & MAPID_BASEMASK) == MAPID_GUNSLINGER) {
+					// For some reason, the client only shows the singular message when it bType is 0 (instead of 1).
+					int btype = (require.spiritball == 1 ? 0 : require.spiritball);
```

Comment:
I was wondering whether we should make a new variable for coins (be it for requirements only or entirely). but since they seem to always force gunslinger to be away of spirit spheres, I guess we are safe..

Reply:
I agree it's a bit confusing that they share the same variable. It might have been implemented this way because I believe even officially they're the same thing (until they realized that there were unexpected interactions between them, through bestowment or card-granted spheres).

If we split them, we need to explicitly take care of all those interactions, and convert between spheres and coins where necessary. Otherwise they can stay as they are, until the day we'll need to have both at once

Reply:
yeah, I agree. guess it is better to leave this way for now. thanks!